abci/server/grpc_server.go

```diff
@@ -46,9 +47,18 @@ func (s *GRPCServer) OnStart() error {
 	types.RegisterABCIApplicationServer(s.server, s.app)
 
 	s.Logger.Info("Listening", "proto", s.proto, "addr", s.addr)
-	go s.server.Serve(s.listener)
+	go func() {
+		if err := s.server.Serve(s.listener); err != nil {
+			errChan <- err
+		}
+	}()
 
-	return nil
+	select {
+	case err = <-errChan:
+		return err
+	default:
+		return nil
+	}
```

Comment:
This is just going to return `nil` immediately. You can either wait for "some time" and then return, or you'll need a different mechanism to pass errors. For now, I'd say just log the error in the goroutine, unless you want to do a bunch of refactoring.